drivers/lis2dh12/lis2dh12.c

```diff
@@ -493,16 +499,23 @@ int lis2dh12_set_fifo(const lis2dh12_t *dev, const lis2dh12_fifo_t *config)
     LIS2DH12_CTRL_REG5_t reg5 = {0};
     LIS2DH12_FIFO_CTRL_REG_t fifo_reg = {0};
 
+    /* cppcheck-suppress unreadVariable; (reason: union type is used for read) */
```

Comment:
@benpicco I'm really sorry to have to add all these "cppcheck-suppress" comments to this code, which make the code unreadable. I wish there was another solution as all these comments are pretty senseless and only there to make "cppcheck" happy.

Reply:
@benpicco Should I revert this commit and suppress the whole file for the moment? If yes, do you know how I can do this in CI? In my local version, I can just add option `-i drivers/lis2dh12/lis2dh12.c`.

Reply:
If `cppcheck` is broken then we should disable it entirely than work around it being broken.

Reply:
#17282 is merged - just rebase and drop the commit :)

Reply:
Done